src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,689 @@
+# Working with Bit Patterns in Clients Last Seen
+
+Monthly active users (MAU) is a windowed metric that requires joining data
+per client across 28 days. Calculating this from individual pings or daily
+aggregations can be computationally expensive, which motivated creation of the
+[`clients_last_seen` dataset](../datasets/bigquery/clients_last_seen/reference.md)
+for desktop Firefox and similar datasets for other applications.
+
+A powerful feature of the `clients_last_seen` methodology is that it doesn't
+record specific metrics like MAU and WAU directly, but rather each row stores
+a history of the discrete days on which a client was active in the past 28 days.
+We could calculate active users in a 10 day or 25 day window just as efficiently
+as a 7 day (WAU) or 28 day (MAU) window. But we can also define completely new
+metrics based on these usage histories, such as various retention definitions.
+
+The usage history is encoded as a "bit pattern" where the physical
+type of the field is a BigQuery INT64, but logically the integer
+represents an array of bits, with each 1 indicating a day where the given clients
+was active and each 0 indicating a day where the client was inactive. This
+article discusses the details of how we represent usage in bit patterns,
+how to extract standard usage and retention metrics,
+and how to build new metrics from them.
+
+## Calculating DAU, WAU, and MAU
+
+The simplest application of usage bit patterns is for calculating metrics in
+backward-looking windows. This is what we do for our canonical _usage_ measures
+DAU, WAU, and MAU.
+
+To decide whether a given client should count towards DAU, WAU, and MAU, we
+need to know how recently that client was active. If the client was seen
+in the past 28 days, they count toward MAU. If they were active in the past 7
+days, the count toward WAU. And only if they were active today do they count
+toward DAU.
+
+The user-facing `clients_last_seen` views present fields like `days_since_seen`
+that extract this information for us from the underlying `days_seen_bits` field,
+allowing us to easily express DAU, WAU, and MAU aggregates like:
+
+```
+SELECT
+  submission_date,
+  COUNTIF(days_since_seen < 28) AS mau,
+  COUNTIF(days_since_seen <  7) AS wau,
+  COUNTIF(days_since_seen <  1) AS dau
+FROM
+  telemetry.clients_last_seen
+WHERE
+  submission_date = '2020-01-28'
+GROUP BY
+  submission_date
+ORDER BY
+  submission_date
+```
+
+Under the hood, `days_since_seen` is calculated using the `bits28_days_since_seen`
+UDF which is explained in more detail later in this article.
+
+Note that the desktop `clients_last_seen` table also has additional bit pattern
+fields corresponding to additional [usage criteria](../metrics/index.md),
+so other variants on MAU can be calculated like:
```

Comment:
Do you want to put forward any comments on what it takes to get new usage critieria? Why these usage criteria?  My understanding is that this criteria is not flexible in production as is, instead we need to make a formal request for each additional usage criteria.